coffeecard/CoffeeCard.Library/Services/AccountService.cs

```diff
@@ -256,6 +248,60 @@ public async Task<bool> RecoverUserAsync(string token, string newPassword)
             await _context.SaveChangesAsync();
             return true;
         }
+        
+        public bool UserExists(string email)
+        {
+            return _context.Users.Any(x => x.Email == email);
+        }
+
+        public void AnonymizeAccount(string email)
+        {
+            var user = _context.Users.First(x => x.Email == email);
+            
+            var claims = new[]
+            {
+                new Claim(ClaimTypes.Email, user.Email), new Claim(ClaimTypes.Name, user.Name),
+                new Claim(ClaimTypes.Role, "verification_token")
+            };
+            var verificationToken = _tokenService.GenerateToken(claims);
+            
+            _emailService.SendVerificationEmailForDeleteAccount(user, verificationToken);
+        }
+        
+        public bool VerifyAnonymization(string token)
```

Comment:
I think the VerifyAnonymization method name is a bit misleading since we do not verify anonymization, but instead, _performs_ the anonymization. From the client, it is seen as a verification but I think method name should reflect what it actually performs

Reply:
The same would also apply to the name of the method called from the controller. It should have a name reflecting that we are making a request to delete an account but doesn't do it in that method call